src/source.rs

```diff
@@ -198,3 +198,52 @@ impl<'a> Todo<'a> {
         })
     }
 }
+
+fn normalize_quote(s: &str) -> String {
+    s
+        .lines()
+        .fold(String::new(), |mut s, l| {
+            let l = l.trim();
+            if !l.is_empty() {
+                s.push_str(l);
+                s.push(' ');
```

Comment:
Sorry I should have commented earlier, you can get rid of the last `.trim().into()` (which does an additional allocation and copy) by pushing a space at the beginning:

```rust
if s.is_empty() {
    s.push(' ');
}
s.push_str(l);
```

Reply:
Ah, nice to know.  Thanks!

Reply:
Emm, that also doesn't quite get there as it introduces a space at the start that also needs to be trimmed. How about this?

```rs
s
        .lines()
        .map(|l| l.trim())
        .filter(|l| !l.is_empty())
        .collect::<Vec<&str>>()
        .join(" ")
```

Reply:
You should only push a space if `s` isn't empty. That will prevent the space at the beginning.

I'm also fine with the `join` route since it's nice and readable but it _does_ introduce an allocation for the `collect::<Vec>`. Not a huge deal though.

Reply:
Oh, I misread your comment. That's really clever.